packages/babel-runtime/helpers/toArray/index.js

```diff
@@ -0,0 +1,14 @@
+var arrayWithHoles = require("@babel/runtime/helpers/arrayWithHoles");
+
+var iterableToArray = require("@babel/runtime/helpers/iterableToArray");
+
+var unsupportedIterableToArray = require("@babel/runtime/helpers/unsupportedIterableToArray");
+
+var nonIterableRest = require("@babel/runtime/helpers/nonIterableRest");
+
+function _toArray(arr) {
+  return arrayWithHoles(arr) || iterableToArray(arr) || unsupportedIterableToArray(arr) || nonIterableRest();
+}
+
+module.exports = _toArray;
```

Comment:
as mentioned on Slack - this won't work because one can write:
```
// node_modules/cast-array/index.js
const toArray = require("@babel/runtime/helpers/toArray")
module.exports = maybeArr => toArray(maybeArr) // it doesn't quite matter what's here, the prior line is important
```
and then one can consume such a package using webpack 5 with:
```
import castArray from 'cast-array'

castArray(document.querySelectorAll('.foo'))
```

Here webpack would consume the first module (`cast-array`) and assume that what it has imported has the shape of `module.exports = { default: toArray }` which would fail loudly at runtime (or at build time? not sure actually).

Reply:
> one can write

This package is only meant to be consumed by generated code, not manually. It webpack throws in that case, or generates invalid code, that's ok. Babel will always inject imports like `require("@babel/runtime/helpers/toArray").default`.

Reply:
Whats the reason for the direct assignment then? Why it exists? When it will be used?

Reply:
So that we can emit this code in Node.js:
```js
import toArray from "@babel/runtime/helpers/toArray"
```
instead of
```js
import toArray_moduleExports from "@babel/runtime/helpers/toArray"
const toArray = toArray_moduleExports.default;
```

Reply:
Could u describe the scenario end to  end?

Reply:
I believe that a problem is that the output consumer might not be a final consumer. The transform-runtime plugin is often used when building packages and im mostly worried about scenarios related to that while i think u are describing the former scenario - correct me if i have misunderstood your intention

Reply:
There are different usages I'm thinking of, and I don't think that whether the file is a pre-compiled package or a compiled source file of the final app matters.

1. **`.mjs` output, used in Node.js**
    ```js
    import toArray from "@babel/runtime/helpers/toArray";
    ``` 
    This will load `@babel/runtime/helpers/toArray/index.cjs`, and it's `module.exports` will be used by Node.js as the value of `toArray`. Internally, Node.js does something like `const toArray = require("@babel/runtime/helpers/toArray/index.cjs");`
1. **`.cjs` output, used in Node.js**
    ```js
    const toArray = require("@babel/runtime/helpers/toArray").default;
    ``` 
    This will load `@babel/runtime/helpers/toArray/index.cjs`, and it's `module.exports.default` will be used as the value of `toArray`.
1. **`.mjs` output, used in [bundler]**
    ```js
    import toArray from "@babel/runtime/helpers/toArray";
    ``` 
    This will load `@babel/runtime/helpers/toArray/index.mjs`, and Webpack will internally use the `default` property of that file's namespace object (`import("@babel/runtime/helpers/toArray/index.mjs").default`).
1. **`.cjs` output, used in [bundler]**
    ```js
    const toArray = require("@babel/runtime/helpers/toArray").default;
    ``` 
    This will load `@babel/runtime/helpers/toArray/index.mjs`, and it's `default` will be used as the value of `toArray` (`import("@babel/runtime/helpers/toArray/index.mjs").default`).

By doing so, Node.js will only load the `.cjs` file and [bundler] will only load the `.mjs` file, and they will never be loaded together.

Reply:
I *think* this indeed might work 👍 Would not bet money on it though 😉 but that's mainly because how the overall story is complicated, not because of the chosen approach